machine/v1beta1/types_azureprovider.go

```diff
@@ -174,6 +174,13 @@ type Image struct {
 	Version string `json:"version"`
 	// ResourceID specifies an image to use by ID
 	ResourceID string `json:"resourceID"`
+	// ThirdPartyImage indicates the image is published by a third party publisher and a Plan
+	// will be generated for it.
+	// +optional
+	ThirdPartyImage bool `json:"thirdPartyImage,omitempty"`
```

Comment:
We avoid booleans in our API because they usually don't age well.

What would this be called if the value was `ThirdParty`?  Is it an `ImageAuthor`? `ImageSource`?  What is the alternative to third party?

Reply:
`ImageSource` would probably be closest. From https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/928 they define it as:
`Add ThirdPartyImage flag to Marketplace image to indicate an image published by a third party publisher.`

We already have a field called `Publisher` so it is not good to call it `ImagePublisher`, etc

Reply:
If it not  published by a third party, who is published by?  microsoft, azure, redhat, someone else?

Reply:
If it's not published by a third party, it's published by Microsoft or one of its cohorts (Canonical/Ubuntu or in a case that is relevant to OpenShift Windows Container the publisher is `WindowsServer`). 

I can't find in-depth documentation on this. [These repo docs](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/docs/book/src/topics/custom-images.md#using-azure-marketplace) are consistent with my reading the code that the distinguishing factor is that "third-party images" need a purchase plan. The cluster api provider has a conditional statement `if third-party marketplace image then generate purchase plan`.

So for our particular use case, when the azure cluster-api provider(CAPZ)  is provisioning a new RHCOS node using a marketplace image, the CAPZ needs to include a purchase plan parameter, but for a first-party Windows Container node, a purchase plan must not be included. 

So we expect the CAPZ to use this API to produce results like this: 
**RHCOS image published by Red Hat**
```go
virtualMachine := &compute.VirtualMachine{
		Plan:     &compute.Plan{
			Publisher: to.StringPtr(image.Publisher), // redhat-rhel
			Name:      to.StringPtr(image.SKU),         // ocp-4
			Product:   to.StringPtr(image.Offer),        // ocp-rhcos-worker
		},
		VirtualMachineProperties: &compute.VirtualMachineProperties{
			StorageProfile: &compute.StorageProfile{
				ImageReference: 	imageReference := &compute.ImageReference{
					Publisher: to.StringPtr(vmSpec.Image.Publisher), // redhat-rhel
					Offer:     to.StringPtr(vmSpec.Image.Offer),           // ocp-rhcos-worker
					Sku:       to.StringPtr(vmSpec.Image.SKU),           // ocp-4
					Version:   to.StringPtr(vmSpec.Image.Version),    // 4.9.7
				},
```

vs

**Windows Server image**
```go
virtualMachine := &compute.VirtualMachine{
		Plan:     nil,
		VirtualMachineProperties: &compute.VirtualMachineProperties{
			StorageProfile: &compute.StorageProfile{
				ImageReference: 	imageReference := &compute.ImageReference{
					Publisher: to.StringPtr(vmSpec.Image.Publisher), // MicrosoftWindowsServer
					Offer:     to.StringPtr(vmSpec.Image.Offer),           // WindowsServer
					Sku:       to.StringPtr(vmSpec.Image.SKU),           // datacenter-core-20h2-with-containers-smalldisk
					Version:   to.StringPtr(vmSpec.Image.Version),    // latest
				},
```